src/js/data.js

```diff
@@ -270,6 +269,17 @@ DataFactory.prototype = {
   }
 }
 
+var DataError = function(message) {
```

Comment:
We should extrapolate this function into a separate file called `errorHandler.js`. Embedding it like this breaks from the current code pattern